File: frontend/src/app/core/helpers/signalr/SignalRMessageResponse.ts
import ConversationId from "../../entities/conversations/keys/ConversationId";
import { MessageType } from "../../entities/messages/enums/MessageType";
import { SignalRFlowMessageResponse } from "./SignalRFlowMessageResponse";

export default interface SignalRMessageResponse {
  content?: string;
  conversationId: ConversationId;
  received: boolean;
  type: MessageType;
  isNotify: boolean;
  createdAt: string;
  flowMessage?: SignalRFlowMessageResponse;
}

File: backend/SprintBusiness/SprintBuisness.Contracts/Whatsapp/Dtos/MessageResponse/SignalRMessageResponse.cs
using SprintBusiness.Domain.Conversations.Keys;
using SprintBusiness.Domain.Messages.Enums;

namespace SprintBuisness.Contracts.Whatsapp.Dtos.MessageResponse
{
    public class SignalRMessageResponse
    {
        public string? Content { get; set; }
        public required ConversationId ConversationId { get; set; }
        public required bool Received { get; set; }
        public required MessageType Type { get; set; }
        public required bool IsNotify { get; set; }
        public required DateTime CreatedAt { get; set; }
        public SignalRFlowMessageResponse? FlowMessage { get; set; }
    }
}
